File: agBa/src/main/java/com/xml/agBa/service/PricelistService.java
package com.xml.agBa.service;

import java.util.List;

import com.xml.agBa.dto.PricelistDTO;
import com.xml.agBa.model.Pricelist;

public interface PricelistService {
	
	PricelistDTO createPricelist(PricelistDTO pricelistDTO);
	List<PricelistDTO> getAllPricelists();
	List<Pricelist> getActivePricelists(Long userId);
	Boolean deletePricelist(Long id);
	Pricelist getPricelistById(Long id);
	Boolean updatePricelist(Long id, PricelistDTO pricelistDTO);
}

File: microservices-front/src/app/services/pricelist.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PricelistDTO } from '../ad/pricelist-create/PricelistDTO';

@Injectable({
  providedIn: 'root'
})
export class PricelistService {

  private baseUrl = 'http://localhost:8083/ad/pricelist';


  constructor(private http: HttpClient) { }

  createNewPricelist(data: PricelistDTO): Observable<any> {
    return this.http.post(this.baseUrl, data);
  }

  updatePricelist(id: number, updateData: any): Observable<any> {
    return this.http.put(this.baseUrl + '/' + id, updateData);
  }

  getPricelistById(id: number): Observable<any> {
    return this.http.get(this.baseUrl + '/' + id);
  }

  getAllPricelists(): Observable<any> {
    return this.http.get(this.baseUrl);
  }

  deletePricelist(id: number): Observable<any> {
    return this.http.delete(this.baseUrl + '/' + id);
  }
}
